src/prefect/orion/orchestration/core_policy.py

```diff
@@ -811,3 +812,32 @@ async def before_transition(
             ),
         )
         return
+
+
+class HandleCancellingScheduledFlowRuns(BaseOrchestrationRule):
+    """Rejects transitions from Scheduled to Cancelling, and sets the state to Cancelled, if
+    the flow run has no associated infrastructure process ID.
+
+    The `Cancelling` state is used to clean up infrastructure. If there is not infrastructure
+    to clean up, we can transition directly to `Cancelled`. Runs that are `AwaitingRetry` are
```

Comment:
Does this comment indicate we want to exclude `AwaitingRetry` named states from this short-circuit?

Reply:
It's currently just a context comment. In theory if there is somehow an AwaitingRetry flow run without an infra PID, we should jump it straight to cancelled too